README.md
# nmr-predictor

  [![NPM version][npm-image]][npm-url]
  [![build status][travis-image]][travis-url]
  [![David deps][david-image]][david-url]
  [![npm download][download-image]][download-url]

NMR chemical shift predictor

## Installation

```
$ npm install nmr-predictor
```

## Example

```js
'use strict';

const predictor = require('..');

const molfile = `Benzene, ethyl-, ID: C100414
  NIST    16081116462D 1   1.00000     0.00000
Copyright by the U.S. Sec. Commerce on behalf of U.S.A. All rights reserved.
  8  8  0     0  0              1 V2000
    0.5015    0.0000    0.0000 C   0  0  0  0  0  0           0  0  0
    0.0000    0.8526    0.0000 C   0  0  0  0  0  0           0  0  0
    1.5046    0.0000    0.0000 C   0  0  0  0  0  0           0  0  0
    2.0062    0.8526    0.0000 C   0  0  0  0  0  0           0  0  0
    3.0092    0.8526    0.0000 C   0  0  0  0  0  0           0  0  0
    1.5046    1.7554    0.0000 C   0  0  0  0  0  0           0  0  0
    0.5015    1.7052    0.0000 C   0  0  0  0  0  0           0  0  0
    3.5108    0.0000    0.0000 C   0  0  0  0  0  0           0  0  0
  1  2  2  0     0  0
  3  1  1  0     0  0
  2  7  1  0     0  0
  4  3  2  0     0  0
  4  5  1  0     0  0
  6  4  1  0     0  0
  5  8  1  0     0  0
  7  6  2  0     0  0
M  END
`;

predictor.fetchProton().then(function () {
    console.log(predictor.proton(molfile));
});
```

## License

  [MIT](./LICENSE)

[npm-image]: https://img.shields.io/npm/v/nmr-predictor.svg?style=flat-square
[npm-url]: https://www.npmjs.com/package/nmr-predictor
[travis-image]: https://img.shields.io/travis/cheminfo-js/nmr-predictor/master.svg?style=flat-square
[travis-url]: https://travis-ci.org/cheminfo-js/nmr-predictor
[david-image]: https://img.shields.io/david/cheminfo-js/nmr-predictor.svg?style=flat-square
[david-url]: https://david-dm.org/cheminfo-js/nmr-predictor
[download-image]: https://img.shields.io/npm/dm/nmr-predictor.svg?style=flat-square
[download-url]: https://www.npmjs.com/package/nmr-predictor


History.md
<a name="0.2.3"></a>
## [0.2.3](https://github.com/cheminfo-js/nmr-predictor/compare/v0.2.2...v0.2.3) (2016-12-16)



<a name="0.2.2"></a>
## [0.2.2](https://github.com/cheminfo-js/nmr-predictor/compare/v0.2.1...v0.2.2) (2016-12-05)



<a name="0.2.1"></a>
## [0.2.1](https://github.com/cheminfo-js/nmr-predictor/compare/v0.1.0...v0.2.1) (2016-12-05)



0.0.1 / HEAD
============

* first release


src/__test__/test2D.js
'use strict';

const NmrPredictor = require('..');
const fs = require('fs');


const db1H = JSON.parse(fs.readFileSync(__dirname + '/../data/h1.json').toString());
const db13C = JSON.parse(fs.readFileSync(__dirname + '/../data/nmrshiftdb2.json').toString());

const molfile = `Benzene, ethyl-, ID: C100414
  NIST    16081116462D 1   1.00000     0.00000
Copyright by the U.S. Sec. Commerce on behalf of U.S.A. All rights reserved.
  8  8  0     0  0              1 V2000
    0.5015    0.0000    0.0000 C   0  0  0  0  0  0           0  0  0
    0.0000    0.8526    0.0000 C   0  0  0  0  0  0           0  0  0
    1.5046    0.0000    0.0000 C   0  0  0  0  0  0           0  0  0
    2.0062    0.8526    0.0000 C   0  0  0  0  0  0           0  0  0
    3.0092    0.8526    0.0000 C   0  0  0  0  0  0           0  0  0
    1.5046    1.7554    0.0000 C   0  0  0  0  0  0           0  0  0
    0.5015    1.7052    0.0000 C   0  0  0  0  0  0           0  0  0
    3.5108    0.0000    0.0000 C   0  0  0  0  0  0           0  0  0
  1  2  2  0     0  0
  3  1  1  0     0  0
  2  7  1  0     0  0
  4  3  2  0     0  0
  4  5  1  0     0  0
  6  4  1  0     0  0
  5  8  1  0     0  0
  7  6  2  0     0  0
M  END
`;

describe('2D prediction', function () {
    this.timeout(1000);
    it('COSY', function (done) {
        var predictor = new NmrPredictor({'H': db1H,'C': db13C});
        var h1 = predictor.protonSync(molfile);
        predictor.twoD(h1, h1, molfile, {minLength: 1, maxLength: 3}).then(prediction => {
            let count = 0;
            prediction.forEach(element => {
                if(element.fromDiaID === "did@`@f\\bbRaih@J@A~dHBIU@"
                    && element.toDiaID === "did@`@fTfUvf`@h@GzP`HeT" && element.pathLength === 3)
                    count++;
                if(element.toDiaID === "did@`@f\\bbRaih@J@A~dHBIU@"
                    && element.fromDiaID === "did@`@fTfUvf`@h@GzP`HeT" && element.pathLength === 3)
                    count++;
                if(element.fromDiaID === "did@`@f\\bbRaih@J@A~dHBIU@"
                    && element.toDiaID === "did@`@fTfYUn`HH@GzP`HeT" && element.pathLength === 3)
                    count++;
                if(element.toDiaID === "did@`@f\\bbRaih@J@A~dHBIU@"
                    && element.fromDiaID === "did@`@fTfYUn`HH@GzP`HeT" && element.pathLength === 3)
                    count++;
                if(element.fromDiaID === "did@`@fTf[Waj@@bJ@_iB@bUP"
                    && element.toDiaID === "did@`@fTeYWaj@@@GzP`HeT" && element.pathLength === 3)
                    count++;
                if(element.toDiaID === "did@`@fTf[Waj@@bJ@_iB@bUP"
                    && element.fromDiaID === "did@`@fTeYWaj@@@GzP`HeT" && element.pathLength === 3)
                    count++;

                if(element.toDiaID === "did@`@fTf[Waj@@bJ@_iB@bUP"
                    && element.fromDiaID === "did@`@fTf[Waj@@bJ@_iB@bUP" && element.pathLength === 2)
                    count++;
                if(element.toDiaID === "did@`@fTeYWaj@@@GzP`HeT"
                    && element.fromDiaID === "did@`@fTeYWaj@@@GzP`HeT" && element.pathLength === 2)
                    count++;
            });

            prediction.length.should.eql(count);
            done();
        });
    });
    it('HSQC', function (done) {
        var predictor = new NmrPredictor({'H': db1H,'C': db13C});
        var c13 = predictor.carbonSync(molfile);
        var h1 = predictor.protonSync(molfile);
        predictor.twoD(h1, c13, molfile, {minLength: 1, maxLength: 4}).then(prediction => {
            prediction.length.should.eql(24);
            done();
        });
    });
});